### PR TITLE
Fix de-duping and property compression, add test (turned off by default), write events in parallel

### PR DESCRIPTION
**Pull request by EmergentOrder**
Fix de-duping and property compression, add test (turned off by default), write events in parallel

### CODE REVIEW

File: core/src/main/scala/io/prediction/core/CleanedDataSource.scala [hunk 1]
```diff
@@ -103,12 +106,29 @@ trait CleanedDataSource {
       } ++ events.filter(!isSetEvent(_))
   }
 
+  //TODO ensure most recent is preserved
   def removePDuplicates(sc: SparkContext, rdd: RDD[Event]): RDD[Event] = {
```
Comment (pferrel) [inline core/src/main/scala/io/prediction/core/CleanedDataSource.scala]: this is pretty important given that the event will be gone prematurely if we don't de-dup to the most recent. Any idea if this is a real problem?

**Comment by pferrel** [comment]
@EmergentOrder do we only have the SimilarItems  test for this?

File: core/src/main/scala/io/prediction/core/CleanedDataSource.scala (hunk 1 above)
Comment (EmergentOrder) [inline core/src/main/scala/io/prediction/core/CleanedDataSource.scala]: Just to clarify eventTime proper is a matching criteria here for de-duping. If 2 events differ by eventTime, they will be considered distinct. However, if they differ only by event Id and/or creationTime, they will be considered dupes. So in regards to creationTime, currently the latest IS preserved. I will add that to the test then remove this comment.

**Comment by EmergentOrder** [comment]
example usage in similar product example, test is in SelfCleaningDataSourceTest, change ignore to test to run

**Pull request merged**